libs/azure-api-client/src/Marketplace/MarketplaceApiClient.php

```diff
@@ -5,28 +5,32 @@
 namespace Keboola\AzureApiClient\Marketplace;
 
 use GuzzleHttp\Psr7\Request;
-use Keboola\AzureApiClient\AzureApiClient;
-use Keboola\AzureApiClient\AzureApiClientFactory;
+use Keboola\AzureApiClient\ApiClient;
+use Keboola\AzureApiClient\ApiClientFactory\AuthenticatedAzureApiClientFactory;
+use Keboola\AzureApiClient\Json;
 use Keboola\AzureApiClient\Marketplace\Model\ActivateSubscriptionRequest;
 use Keboola\AzureApiClient\Marketplace\Model\ResolveSubscriptionResult;
 use Keboola\AzureApiClient\Marketplace\Model\Subscription;
 
 class MarketplaceApiClient
 {
     public function __construct(
-        private readonly AzureApiClient $azureApiClient,
+        private readonly ApiClient $apiClient,
     ) {
     }
 
-    public static function create(AzureApiClientFactory $clientFactory): self
+    public static function create(AuthenticatedAzureApiClientFactory $clientFactory): self
```

Comment:
Tyjo tole mě přijde úplně nahovno, abych vytvořil MarketPlaceApiClient, tak musím vytvořit AuthenticatedAzureApiClientFactory, abych mohl vytvořit to, tak musím vytvořit GuzzleClientFactory a AuthenticatorFactory a to jsem ještě nezačal nic dělat. Místo to, aby se to zjednodušilo, tak je to čím dál složitejší.

````
$client = new Aws\DynamoDb\DynamoDbClient([
    'region'  => 'us-west-2',
    'version' => 'latest',
]);
```` ?

Reply:
Jo, je to zbytecne slozity. Aby to preslo pres Infection, musel jsem toho udelat vic "testovatelnyho" a pokryt to a nejprimejsi cesta k tomu je to proste rozlozit na vic komponent. Pripravil jsem ted https://github.com/keboola/platform-libraries/pull/63, kde se odjebala ta GuzzleClientFactory. Jeste bych se rad zbavil toho `Plain` vs `Authenticated` ApiClient, cimz bysme se mohli dostat snad skoro k examplu vys, ale jeste si na to chci dva/tri vecery sednou. Bude to takle jako future promise stacit? Nebo to vidis jako blocker a mam si tonejak zaprioritizovat?